### PR TITLE
fix: correct parallel computation

### PR DESCRIPTION
**Pull request by chokobole**
# Description

This PR fixes the bug related to openmp.

### CODE REVIEW

**Review by ashjeong — APPROVED** [review]
LGTM

**Review by Insun35 — APPROVED** [review]
LGTM

**Review by dongchangYoo — APPROVED** [review]
LGTM

File: tachyon/math/polynomials/univariate/lagrange_interpolation.h [hunk 1]
```diff
@@ -36,48 +36,51 @@ bool LagrangeInterpolate(
...
+      for (size_t k = k_start; k > 0; --k) {
+        coeffs[k] = coeffs[k - 1] - points[j] * coeffs[k];
       }
+      ++k_start;
+      coeffs[0] *= -points[j];
     }
-  });
+
+    // P(X) = ∑ᵢ (P(Xᵢ) * Dᵢ * Nᵢ(X))
+    //      = ∑ᵢ (P(Xᵢ) * Dᵢ * (cₙ₋₁Xⁿ⁻¹ + cₙ₋₂Xⁿ⁻² + ... + c₁X¹ + c₀X⁰))
```
Comment (TomTaehoonKim) [inline tachyon/math/polynomials/univariate/lagrange_interpolation.h:77]:
```suggestion
    // P(X) = ∑ᵢ(P(Xᵢ) * Dᵢ * Nᵢ(X))
    //      = ∑ᵢ(P(Xᵢ) * Dᵢ * (cₙ₋₁Xⁿ⁻¹ + cₙ₋₂Xⁿ⁻² + ... + c₁X¹ + c₀X⁰))
```

File: tachyon/math/polynomials/univariate/lagrange_interpolation.h [hunk 2]
```diff
@@ -36,48 +36,51 @@ bool LagrangeInterpolate(
...
+  base::Parallelize(denoms, [&points](absl::Span<F> chunk, size_t chunk_offset,
+                                      size_t chunk_size) {
+    size_t i = chunk_offset * chunk_size;
+    for (F& denom : chunk) {
       for (size_t j = 0; j < points.size(); ++j) {
-        if (j != i) {
-          chunk[i] *= (points[i] - points[j]);
-        }
+        if (i == j) continue;
+        denom *= (points[i] - points[j]);
```
Comment (TomTaehoonKim) [inline tachyon/math/polynomials/univariate/lagrange_interpolation.h:48]: It's same as below right?
```c++
        if (i == j) {
          denom *= (points[i] - points[j]);
        }
```

**Review by TomTaehoonKim — CHANGES_REQUESTED** [review]

File: tachyon/math/polynomials/univariate/lagrange_interpolation.h (hunk 2 above)
Comment (chokobole) [reply tachyon/math/polynomials/univariate/lagrange_interpolation.h:48]: Yeap!

**Review by TomTaehoonKim — APPROVED** [review]
LGTM

**Comment by sengulatik66** [comment]
LGTM

**Pull request merged**